stl/inc/algorithm

```diff
@@ -877,7 +877,7 @@ _NODISCARD _CONSTEXPR20 _FwdIt search_n(
                 _Iter_diff_t<_FwdIt> _Count1 = _Count_diff;
                 auto _UMid                   = _UFirst;
 
-                while (_UOld_first != _UFirst && _Pred(_UFirst[-1], _Val)) { // back up over any skipped prefix
+                while (_UOld_first != _UFirst && _Pred(*(_UFirst - 1), _Val)) { // back up over any skipped prefix

```

Comment:
Stupid ?: Should we `_Prev_iter` here for iterators like those of `deque` which are cheaper to decrement than to subtract?

Reply:
I don't see such a difference for unchecked deque iterators - can you show me the code that's slower?

Reply:
They just store offsets:
https://github.com/microsoft/STL/blob/fd04f77dd217a7502c77e0cbf45e24fcb84ed17c/stl/inc/deque#L75-L78
https://github.com/microsoft/STL/blob/fd04f77dd217a7502c77e0cbf45e24fcb84ed17c/stl/inc/deque#L96-L104

Reply:
You're right, our deque just stores an offset in the iterators so they are equivalent: https://gcc.godbolt.org/z/Uwc3QD

libstdc++'s deque is cheaper to --, so containers like this are presumably possible for program-defined iterator types: https://gcc.godbolt.org/z/W23swg

Reply:
I've filed #465 because I believe that's separate from this correctness issue. Thanks for the suggestion!

Reply:
Sounds good to me